File: Q5/python/ThePalindrome.py

class ThePalindrome:
    def __init__(self):
        pass

    def find(self, s : str):
        count = 1
        for i in range(0,s.__len__()):
            count = i
            bDiff = False
            for j in range(0,s.__len__()):
                if j+1 == int(s.__len__()/2) or i + j >= s.__len__():
                    break

                if s[j + i] != s[s.__len__()-1-j]:
                    bDiff = True
                    break


            if not bDiff:
                break

        return s.__len__() + count


print(ThePalindrome().find("abab"))
print(ThePalindrome().find("abacaba"))
print(ThePalindrome().find("qwerty"))
print(ThePalindrome().find("abdfhdyrbdbsdfghjkllkjhgfds"))
print(ThePalindrome().find("abcccc"))

File: Q5/cpp/ThePalindrome.cpp
#include <string>
#include <iostream>

using namespace std;

class ThePalindrome
{
public:
    int find(string s)
    {
        int count = 1;
        for(int i = 0 ; i < s.size(); i++)
        {
            count = i;
            bool bDiff = false;
            for(int j = 0; j < s.size() ; j++)
            {
                if(s[j + i] != s[s.size()-1-j])
                {
                    bDiff = true;
                    break;
                }            

                if(j+1 == (int)s.size()/2)
                {
                    break;
                }
            }

            if(!bDiff)
                break;
        }

        return s.size() + count;
    }
};

int main()
{
    cout<<ThePalindrome().find("abab")<<endl;
    cout<<ThePalindrome().find("abacaba")<<endl;
    cout<<ThePalindrome().find("qwerty")<<endl;
    cout<<ThePalindrome().find("abdfhdyrbdbsdfghjkllkjhgfds")<<endl;
    cout<<ThePalindrome().find("abcccc")<<endl;
    return 0;
}